docs/hub/how-to-downstream.md

```diff
@@ -0,0 +1,55 @@
+* Download all the files in a repository.
+
+## `hf_hub_url`
+
+Use `hf_hub_url` to retrieve the URL of a specific file to download by providing a `filename`.
+
+![/docs/assets/hub/repo.png](/docs/assets/hub/repo.png)
+
+```python
+>>> from huggingface_hub import hf_hub_url
+>>> hf_hub_url(repo_id="lysandre/arxiv-nlp", filename="config.json")
+'https://huggingface.co/lysandre/arxiv-nlp/resolve/main/config.json'
+```
+
+Specify a particular file version by providing the file revision. The file revision can be a branch, a tag, or a commit hash. If you choose to use the commit hash, it must be the full-length hash instead of a 7-character commit hash.
+
+```python
+>>> hf_hub_url(repo_id="lysandre/arxiv-nlp", filename="config.json", revision="877b84a8f93f2d619faa2a6e514a32beef88ab0a")
+'https://huggingface.co/lysandre/arxiv-nlp/resolve/877b84a8f93f2d619faa2a6e514a32beef88ab0a/config.json'
+```
+
+## `cached_download`
+
+The `cached_download` method is useful for downloading and caching a file on your local disk. Once it is stored in your cache, you don't have to redownload the file the next time you use it. This is a hands-free solution for staying up to date with new file versions. When a downloaded file is updated in the repository, `huggingface_hub` will automatically download and store it for you.
+
+Begin by retrieving your file URL with `hf_hub_url`, and then pass the specified URL to `cached_download` to download the file:
+
+```python
+>>> from huggingface_hub import hf_hub_url, cached_download
+>>> config_file_url = hf_hub_url("lysandre/arxiv-nlp", filename="config.json")
+>>> cached_download(config_file_url)
+'/home/lysandre/.cache/huggingface/hub/bc0e8cc2f8271b322304e8bb84b3b7580701d53a335ab2d75da19c249e2eeebb.066dae6fdb1e2b8cce60c35cc0f78ed1451d9b341c78de19f3ad469d10a8cbb1'
+```
+
+The `hf_hub_url` and `cached_download` functions work hand in hand to download a file. In fact, this is exactly how the `hf_hub_download` function from the tutorial works! `hf_hub_download` is simply a wrapper that calls both `hf_hub_url` and `cached_download`.
+
+## `snapshot_download`
+
+The `snapshot_download` method downloads an entire repository at a given revision. Like the `cached_download` function, all downloaded files are cached on your local disk. However, even if only a single file is updated, the entire repository will be redownloaded.
+
+Download a whole repository as shown in the following:
+
+```python
+>>> from huggingface_hub import snapshot_download
+>>> snapshot_download(repo_id="lysandre/arxiv-nlp")
+'/home/lysandre/.cache/huggingface/hub/lysandre__arxiv-nlp.894a9adde21d9a3e3843e6d5aeaaf01875c7fade'
+```
+
+In general, it is usually better to manually download files with `hf_hub_download` (if you already know which files you need) to avoid redownloading an entire repository. `snapshot_download` is helpful when your library's downloading utility is a helper, and unaware of which files need to be downloaded.
```

Comment:
Do we have "Tip" boxes in these docs? This would be a good candidate for one of them IMO